src/components/picker/helpers/usePickerSearch.tsx

```diff
@@ -3,33 +3,32 @@ import _ from 'lodash';
 import {PickerProps} from '../types';
 import {getItemLabel as getItemLabelPresenter, shouldFilterOut} from '../PickerPresenter';
 
-type UsePickerSearchProps = Pick<PickerProps, 'showSearch' | 'onSearchChange' | 'children' | 'getItemLabel'>;
+type UsePickerSearchProps = Pick<PickerProps, 'showSearch' | 'onSearchChange' | 'children' | 'getItemLabel' | 'items'>;
 
 const usePickerSearch = (props: UsePickerSearchProps) => {
-  const {showSearch, onSearchChange, children, getItemLabel} = props;
+  const {showSearch, onSearchChange, children, getItemLabel, items} = props;
   const [searchValue, setSearchValue] = useState('');
 
-  const filteredChildren = useMemo(() => {
+  const filterItems = (items: any, getItemLabelFunction: any) => {
     if (showSearch && !_.isEmpty(searchValue)) {
-      // @ts-expect-error need to fix children type
-      return _.filter(children, child => {
-        // @ts-expect-error need to fix children type to be based on PickerItemProps
-        const {label, value, getItemLabel: childGetItemLabel} = child.props;
-        const itemLabel = getItemLabelPresenter(label, value, childGetItemLabel || getItemLabel);
+      return _.filter(items, item => {
+        const {label, value, getItemLabel} = item.props || item;
+        const itemLabel = getItemLabelPresenter(label, value, getItemLabel || getItemLabelFunction);
         return !shouldFilterOut(searchValue, itemLabel);
       });
     }
+    return items;
+  };
 
-    return children;
-  }, [showSearch, searchValue, children]);
+  const filteredItems = useMemo(() => filterItems(children || items, getItemLabel), [showSearch, searchValue, items, children]);
```

Comment:
Lets move everything into the `useMemo`?

Reply:
@M-i-k-e-l After going over it few times, the are many type errors when moving the `filterItems` function into `filteredItems` and sending items as `children || items` the type we are getting back from the function is problemtic.
I'm not sure if it's worth time to go into it since it will be solved after the children migration finish.
WDYT ?

Reply:
Lets take a look after you've resolved the other issues.

Reply:
I wrapped the filterItems with `useCallback`